transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -606,6 +602,15 @@ protected void run() {
         }
     }
 
+    // returns true if selectCnt should be reset
+    private boolean selectReturnPrematurely(int selectCnt, boolean ranTasks, int strategy) {
+        if (selectCnt > MIN_PREMATURE_SELECTOR_RETURNS && logger.isDebugEnabled()) {
+            logger.debug("Selector.select() returned prematurely {} times in a row for Selector {}.",
+                selectCnt - 1, selector);
+        }
+        return ranTasks || strategy > 0;
```

Comment:
hmm... seems like the semantic changed. I think it should only log if `ranTask || strategy > 0` just like before

Reply:
Ouch, let me change that.